### PR TITLE
Export user-specified transform in C API.

### PR DESCRIPTION
**Pull request by jpap**

### CODE REVIEW

**Comment by jpap** [comment]
I needed this for a SVG having canvas size `C` and RESVG_FIT_TO_ZOOM with a scaling factor `S`.  When rendered into a pixmap of size smaller than `C*S`, I had expected the render to be centered in that pixmap: instead it was just rendered to origin in the top-left.

By using a translating transform, I can get the SVG to be scaled up and centered in the above pixmap.

File: c-api/lib.rs [hunk 1]
```diff
@@ -592,14 +604,24 @@ pub extern "C" fn resvg_render(
     let pixmap: &mut [u8] = unsafe { std::slice::from_raw_parts_mut(pixmap as *mut u8, pixmap_len) };
     let pixmap = tiny_skia::PixmapMut::from_bytes(pixmap, width, height).unwrap();
 
-    resvg::render(&tree.0, fit_to.to_usvg(), tiny_skia::Transform::default(), pixmap).unwrap()
+    let skia_xf = tiny_skia::Transform::from_row(
```
Comment (RazrFalcon) [inline c-api/lib.rs:607]: Prefer:

```rust
impl resvg_transform {
    #[inline]
    fn to_tiny_skia(&self) -> tiny_skia::Transform {
        tiny_skia::Transform::from_row(
            self.a as f32, 
            self.b as f32, 
            self.c as f32, 
            self.d as f32, 
            self.e as f32, 
            self.f as f32,
        );
    }
}
```

and then `transform.to_to_tiny_skia()`, just like with `resvg_fit_to`

File: c-api/resvg.h [hunk 2]
```diff
@@ -166,6 +166,11 @@ typedef struct resvg_transform {
     double f;
 } resvg_transform;
 
+/**
+ * @brief Returns an identity transform.
+ */
+resvg_transform resvg_transform_default();
```
Comment (RazrFalcon) [inline c-api/resvg.h:172]: I think we should name it `resvg_transform_identity` to be more clear.

File: c-api/ResvgQt.h [hunk 3]
```diff
@@ -530,6 +530,7 @@ class ResvgRenderer {
      */
     QImage renderToImage(const QSize &size = QSize()) const
     {
+        resvg_transform xf = resvg_transform_default();
```
Comment (RazrFalcon) [inline c-api/ResvgQt.h:533]: This function should accept `QTransform` then, but I guess I can add this myself.
At least rename `xf` to `ts`.

**Review by RazrFalcon — CHANGES_REQUESTED** [review]

File: c-api/resvg.h (hunk 2 above)
Comment (jpap) [reply c-api/resvg.h]: I have made the requested change, but the the "default" name was chosen to mirror the same name you chose in tiny_skia.  I don't mind either way...

File: c-api/ResvgQt.h (hunk 3 above)
Comment (jpap) [reply c-api/ResvgQt.h]: Please go ahead and refactor to incorporate `QTransform`.

File: c-api/ResvgQt.h [hunk 4]
```diff
@@ -441,15 +441,15 @@ class ResvgRenderer {
      */
     QRect viewBox() const
     {
-        return viewBoxF().toRect();
+        return viewBots().toRect();
```
Comment (RazrFalcon) [inline c-api/ResvgQt.h:444]: ?!

**Review by RazrFalcon — CHANGES_REQUESTED** [review]

File: c-api/ResvgQt.h (hunk 4 above)
Comment (jpap) [reply c-api/ResvgQt.h]: Thanks for catching this: was a slip in a case-insensitive search and replace.

**Review by RazrFalcon — APPROVED** [review]

**Comment by RazrFalcon** [comment]
Thanks!

**Pull request merged**